react-native-config.podspec

```diff
@@ -20,12 +20,12 @@ Pod::Spec.new do |s|
   s.script_phase = {
     name: 'Config codegen',
     script: %(
-set -ex
-HOST_PATH="$SRCROOT/../.."
-"${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig/BuildDotenvConfig.rb" "$HOST_PATH" "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig"
-),
+      set -ex
+      HOST_PATH="$SRCROOT/../.."
+      "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig/BuildDotenvConfig.rb" "$HOST_PATH" "${PODS_TARGET_SRCROOT}/ios/ReactNativeConfig"
+    ),
     execution_position: :before_compile,
-    input_files: ['$PODS_TARGET_SRCROOT/ios/ReactNativeConfig/BuildDotenvConfig.rb'],
+    input_files: ['$PODS_TARGET_SRCROOT/ios/ReactNativeConfig/BuildDotenvConfig.rb', '$SRCROOT/../../.env', '$ENVFILE'],
```

Comment:
Maybe need to add `'$SRCROOT/../../$ENVFILE'` for relative file name.

Reply:
It's a bit tricky 😁
**$ENVFILE** is used for the case when the `env` file is set via **build_settings**, like so (in your Podfile):

```
ENVFILES = {
  'Debug' => '$(PODS_ROOT)/../../.env.debug',
  'Release' => '$(PODS_ROOT)/../../.env.production',
}
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      if target.name == 'react-native-config'
        config.build_settings['ENVFILE'] = ENVFILES[config.name]
      end
    end
  end
end
```

Reply:
Ah, I don't use the "alternative" path as described here: https://github.com/lugg/react-native-config?tab=readme-ov-file#ios-1

But instead define ENVFILE as environment variable before running the react native scripts from the command line:

In the `scripts` section of `package.json`:
```json
"sim.prod": "cross-env ENVFILE=.env.production  react-native run-ios --no-packager --simulator 'iPhone 15 Pro (17.2)'",
```

Since the Ruby files in this package are reading the environment variables.

I guess I could switch that to use a scheme arg on the CLI instead, but others may also make the above assumption that setting the ENVFILE env var works.

Reply:
In this case, I think the following is better:
```json
"sim.prod": "cross-env ENVFILE=$(pwd)/.env.production react-native run-ios --no-packager --simulator 'iPhone 15 Pro (17.2)'"
```

Why:
To avoid mixing concerns. The Xcode project should not define where the ENVFILE is located (except for the default location like **'$SRCROOT/../../.env'**).